packages/next/src/build/compile-config.ts

```diff
@@ -0,0 +1,36 @@
+import { join } from 'path'
+import { mkdir, readFile, writeFile } from 'fs/promises'
+import { transform } from './swc'
+
+export async function compileConfig({
+  configPath,
+  cwd,
+  log,
```

Comment:
IMHO, you can import the log module from `next/src/build/output/log` directly.

Reply:
It was intended since the `curLog` was pre-defined here:

https://github.com/vercel/next.js/blob/22b1497f21f07a364cbb22fba1f3f9951b770a56/packages/next/src/server/config.ts#L925-L931

and `loadEnvConfig` had this curLog passed as params

https://github.com/vercel/next.js/blob/22b1497f21f07a364cbb22fba1f3f9951b770a56/packages/next/src/server/config.ts#L933

so I passed it to compileConfig to not redeclare the log + for consistency.

Reply:
Nice catch! Then the type of the log should not be `any` (It should be `Pick<typeof import('./output/log'), 'error'>` then).

Reply:
Thank you!